plugins/janus_videoroom.c

```diff
@@ -990,6 +990,7 @@ void janus_videoroom_destroy_session(janus_plugin_session *handle, int *error) {
 	if(!g_atomic_int_get(&session->destroyed)) {
 		/* Any related WebRTC PeerConnection is not available anymore either */
 		janus_videoroom_hangup_media(handle);
+		handle->plugin_handle = NULL;
```

Comment:
The reason why I'm resetting this before that is to make sure that as soon as a session has been marked as destroyed, any further attempt to use it is denied, even before the session is freed. I had thought about a mutex for that pointer, but that would force us to lock/unlock even on potentially very frequent callbacks like `incoming_rtp`, something I'd rather avoid. That's what I mentioned by "giving this more thought" in my comment before, we'll have to evaluate what the safest approach is without harming performance too much. If it turns out a lock is the only way then I guess we can do that.